lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2.cpp

```diff
@@ -363,7 +362,7 @@ bool MMU2::rx_str_P(const char *str) {
 
     str += len;
 
-    while (len--) {
+    while (len-- && i < sizeof(rx_buffer)) {
```

Comment:
Have you noticed that the i is being decreased inside the loop?

Reply:
yes. if i is 0 and the loop continues, i is decreased and will have the value 0xFF on the next run (being uint8_t), which is bigger than the size of rx_buffer.

Reply:
And have you also noticed that `i` always will be larger than `len` because of the theck two lines above?

Reply:
yep, you're right.